src/ddmd/lexer.d

```diff
@@ -2096,17 +2103,19 @@ class Lexer
         auto sbufptr = cast(const(char)*)stringbuffer.data;
         TOK result;
         bool isOutOfRange = false;
-        t.floatvalue = CTFloat.parse(sbufptr, &isOutOfRange);
+        t.floatvalue = (isWellformedString ? CTFloat.parse(sbufptr, &isOutOfRange) : CTFloat.zero);
```

Comment:
DMD's `CTFloat.parse()` uses the host C runtime (`strtold()`), which returns 0 for invalid literals.